tests/meta-mender-toradex-nxp-ci/recipes-bsp/u-boot/files/colibri-imx7/fw_env.config

```diff
@@ -0,0 +1,10 @@
+# Configuration file for fw_(printenv/setenv) utility.
+# Up to two entries are valid, in this case the redundant
+# environment sector is assumed present.
+# Notice, that the "Number of sectors" is not required on NOR and SPI-dataflash.
+# Futhermore, if the Flash sector size is ommitted, this value is assumed to
+# be the same as the Environment size, which is valid for NOR and SPI-dataflash
+
+# MTD device name       Device offset   Env. size       Flash sector size       Number of sectors
+/dev/mtd3               0x00000000      0x00020000      0x20000                 4
+/dev/mtd3               0x00020000      0x00020000      0x20000                 4
```

Comment:
Thanks for spotting this. Colibri iMX7 is set up with ENV_SIZE == ENV_SECT_SIZE (128kB). I believe ENV_RANGE will default to ENV_SIZE if not defined (and it's not defined for mx7 either). Redundant copy is currently set up to start right after the main one.

I wouldn't want to change the partitioning just for Mender. I suppose we could push back the redundant environment by another 128kB, so it would start at 0x3c0000 (or 0x40000 in /dev/mtd3), and set number of sectors to 2. ENV_SIZE remains unchanged, ENV_RANGE becomes 2*ENV_SIZE (to be added to clibri_imx7.h). 

With the mender view on things, things get a little complicated though. The setup code assumes that it knows better and sets up the redundant environment right after the main one, at MENDER_BOOTENV_SIZE distance (128kB). In theory I could manipulate MENDER_BOOTENV_SIZE to be 256kB, this would give the necessary space for ENV_RANGE, and just have ENV_SIZE set to 128kB. However, then there's a check for MENDER_BOOTENV_SIZE == ENV_SIZE, which will not be true anymore.